src/python/dxpy/app_builder.py

```diff
@@ -539,6 +539,15 @@ def upload_applet(src_dir, uploaded_resources, check_name_collisions=True, overw
             applet_spec["runSpec"]["code"] = code_fh.read()
             del applet_spec["runSpec"]["file"]
 
+    # If this is a cluster applet, inline all provided bootstrapScript code
+    # NOTE: assumes bootstrapScript is always provided as a filename
+    if "systemRequirements" in applet_spec["runSpec"]:
+        sys_reqs = applet_spec["runSpec"]["systemRequirements"]
+        for entry_point in sys_reqs:
+            if "clusterSpec" in sys_reqs[entry_point]:
+                with open(os.path.join(src_dir, sys_reqs[entry_point]["clusterSpec"]["bootstrapScript"])) as code_fh:
+                    sys_reqs[entry_point]["clusterSpec"]["bootstrapScript"] = code_fh.read()
```

Comment:
We should try/catch here if the file doesn't exist or cannot be accessed.

Reply:
Also, we perform some syntax validation on the source code of the app so maybe we want to do it for this file as well?